ghc/riscv64.patch

```diff
@@ -1,69 +1,77 @@
-+            'ffc73819465dee34435b82b59f04488fd49163466814538121dc8855e3fee4ab23d9085dac108292022f3bffbc45b84cb744428ce42fe21214437b72010aae77'
-+            '8c1a3c7c99feffeb25c44cdeb7f40161c72e034fc2608ec92703a7bbc09e60ab4cd0899175ca20c126556762542509691d57e2a80d5430173c0a827bf51db9c7'
-+            '1e31ced0a68d3782d4bff9105ede96354ac8a97157625449b82006ed5bb1503b50d29114f5101461d999c31a307d8161379ceae0df0ddb33ab1b47cf3918a1ab'
-+            '09d8de013c3d27d77f13cbb78a4eb653121df6716cbd39b50af48e9664ce057377377c44a7243a3c2fdb2b4a48a92dd25f0723fae62597504a79d4629435ba9c')
- 
- prepare() {
-   # Need to extract this tarball with a UTF-8 locale instead of a chroot's "C"
-@@ -37,11 +55,14 @@ prepare() {
+             '3bdbd05c4a2c4fce4adf6802ff99b1088bdfad63da9ebfc470af9e271c3dd796f86fba1cf319d8f4078054d85c6d9e6a01f79994559f24cc77ee1a25724af2e6')
+@@ -47,15 +61,25 @@ prepare() {
  
    cd ghc-$pkgver
  
--  # https://gitlab.haskell.org/ghc/ghc/-/issues/19976
--  patch -p1 -i ../a02fbadaf59521b5f1af3f05b45933b245093531.patch
-+  for f in "$srcdir"/*.patch; do
-+    patch -Np1 -i "$f"
-+  done
+-  patch -p1 -i ../0001-Replace-more-autotools-obsolete-macros-19189.patch
+-  patch -p1 -i ../0002-configure-fix-the-use-of-some-obsolete-macros-19189.patch
+-  patch -p1 -i ../0003-llvmGen-Accept-range-of-LLVM-versions.patch
+-  patch -p1 -i ../0004-Fix-autoconf-after-6d6edb1bbb.patch
+-  patch -p1 -i ../0005-Set-min-LLVM-version-to-9-and-make-version-checking-.patch
+-  patch -p1 -i ../0006-Optimiser-Correctly-deal-with-strings-starting-with-unicode.patch
++  patch -Np1 -i ../0001-Replace-more-autotools-obsolete-macros-19189.patch
++  patch -Np1 -i ../0002-configure-fix-the-use-of-some-obsolete-macros-19189.patch
++  patch -Np1 -i ../0003-llvmGen-Accept-range-of-LLVM-versions.patch
++  patch -Np1 -i ../0004-Fix-FIND_LLVM_PROG-arguments-after-6d6edb1bbb.patch
++  patch -Np1 -i ../0005-Set-min-LLVM-version-to-9-and-make-version-checking-.patch
++  patch -Np1 -i ../0006-Optimiser-Correctly-deal-with-strings-starting-with-.patch
++  patch -Np1 -i ../0007-Limit-upper-version-of-Happy-for-ghc-9.2-18620.patch
++  patch -Np1 -i ../0008-Check-for-libatomic-dependency-for-atomic-operations.patch
++  patch -Np1 -i ../0009-Implement-riscv64-LLVM-backend.patch
++  patch -Np1 -i ../0010-Enable-tables-next-to-code-for-riscv64.patch
++  patch -Np1 -i ../0011-Stop-leaking-defunct-llc-processes.patch
++  patch -Np1 -i ../0012-configure-Fix-redundant-argument-warning-from-no-pie.patch
++  patch -Np1 -i ../0013-Allow-building-registerised-compiler-with-unregister.patch
  
    cp mk/build.mk{.sample,}
--  sed -i '1iBuildFlavour = perf' mk/build.mk
-+  sed -i '1iBuildFlavour = perf-llvm' mk/build.mk
-+
-+  ./boot
- }
+   sed -i '1iBuildFlavour = perf-llvm' mk/build.mk
++  cat >>mk/build.mk <<EOF
++CONF_CC_OPTS_STAGE0 += -Wno-unused-label -Wno-unused-but-set-variable -Wno-unused-variable
```

Comment:
This is only meant to suppress excessive compiler warnings when building with an unregisterised compiler (i.e. `ghc-bootstrap`). It is not required at all in subsequent builds.

It could be a little bit cleaner if we just enable LLVM backend for unregisterised compilers with a one-line patch.